image_ipm/image_ipm/ipm.py

```diff
@@ -0,0 +1,164 @@
+# Copyright (c) 2022 Hamburg Bit-Bots
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import cv2
+from cv_bridge import CvBridge
+from ipm_library.ipm import IPM
+from ipm_library.utils import create_horizontal_plane
+import numpy as np
+from numpy.lib import recfunctions as rfn
+import rclpy
+from rclpy.duration import Duration
+from rclpy.executors import MultiThreadedExecutor
+from rclpy.node import Node
+from sensor_msgs.msg import CameraInfo, Image, PointCloud2, PointField
+from sensor_msgs_py.point_cloud2 import create_cloud
+from std_msgs.msg import Header
+import tf2_ros as tf2
+
+cv_bridge = CvBridge()
+
+
+class ImageIPM(Node):
+
+    def __init__(self) -> None:
+        super().__init__('image_ipm')
+        # We need to create a tf buffer
+        self.tf_buffer = tf2.Buffer(cache_time=Duration(seconds=30.0))
+        self.tf_listener = tf2.TransformListener(self.tf_buffer, self)
+
+        # Create an IPM instance
+        self.ipm = IPM(self.tf_buffer)
+
+        # Declare params
+        self.declare_parameter('output_frame', 'base_footprint')
+        self.declare_parameter('type', 'mask')
+        self.declare_parameter('scale', 1.0)
+
+        # Subscribe to camera info
+        self.create_subscription(CameraInfo, 'camera_info', self.ipm.set_camera_info, 1)
+
+        # Create processing pipeline
+
+        # Masks
+        mask_publisher = self.create_publisher(PointCloud2, 'mask_relative_pc', 1)
+
+        self.create_subscription(
+            Image,
+            'mask_in_image',
+            lambda msg: mask_publisher.publish(self.map_mask(msg)),
+            1)
+
+    def map_mask(self, msg: Image) -> PointCloud2:
```

Comment:
As this node also supports colored images, this should be reflected in the naming scheme.